ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsUtils.kt

```diff
@@ -35,8 +36,7 @@ internal suspend fun HttpRequestData.toRaw(callContext: CoroutineContext): Reque
     return buildObject {
         method = this@toRaw.method.value
         headers = jsHeaders
-        redirect = RequestRedirect.FOLLOW
-
+        redirect = attributes.getOrNull(AttributeKey<String>("redirect")) ?: RequestRedirect.FOLLOW
```

Comment:
How about adding a `getOrDefault` function to `Attributes`?
`fun <T : Any> getOrDefault(key: AttributeKey<T>, default: T): T = getOrNull(key) ?: default`